index.html

```diff
@@ -65,9 +60,15 @@
                 ])
                 const [active, setActive] = useState(0)
 
-                const [currentContent, setCurrentContent] = useState(
-                    documents.find((d) => d.id == active).content,
-                )
+                const getCurrentContent = () => documents.find((d) => d.id === active).content;
+                const setCurrentContent = (newContent) => {
+                    setDocuments((docs) => docs.map(doc => {
```

Comment:
We need to use `map` here to create a new object instance for the changed document(s) because otherwise React doesn't react on the change.

```javascript
docs.find(...).content = newContent;
```

keeps the reference (pointer if you will) in `docs` the same, so React/`setDocuments` does not realize that something has changed.

Reply:
```javascript
docs = [{ id: 42, content: 'test' }];
newDocs = docs;
console.log('docs === newDocs:', docs === newDocs); // => true

newDocs[0].content = 'something else';
console.log('docs === newDocs:', docs === newDocs); // => true
console.log('docs[0] === newDocs[0]:', docs[0] === newDocs[0]); // => true

newDocs = docs.map(doc => doc);
console.log('docs === newDocs:', docs === newDocs); // => false
```